packages/appsync-modelgen-plugin/src/visitors/appsync-visitor.ts

```diff
@@ -498,6 +507,41 @@ export class AppSyncModelVisitor<
     return plural(model.name);
   }
 
+  /**
+   * Add timestamp fields createdAt, updatedAt(or equivalent fields) to model fields
+   * @param model
+   */
+  protected addTimestampFields(model: CodeGenModel, directive: CodeGenDirective): void {
+    const target = this.config.target;
+    if (target === 'javascript' || target === 'dart' || target === 'metadata') {
+      return;
+    }
+    if (directive.name !== 'model') {
+      return;
+    }
+    const timestamps = directive.arguments.timestamps;
+    const createdAtField: CodeGenField = {
+      name: timestamps?.createdAt || DEFAULT_CREATED_TIME,
+      directives: [],
+      type: 'AWSDateTime',
+      isList: false,
+      isNullable: true,
+      isReadOnly: true,
+    };
+    const updatedAtField: CodeGenField = {
+      name: timestamps?.updatedAt || DEFAULT_UPDATED_TIME,
+      directives: [],
+      type: 'AWSDateTime',
+      isList: false,
+      isNullable: true,
+      isReadOnly: true,
+    };
+    addFieldToModel(model, createdAtField);
+    //updated field will always be override
```

Comment:
Why is the updated field always overridden?

Reply:
I just tested the graphql transformer by the latest version of CLI and found out that updated field had the same behavior as created field, which means they will both be added to the input variables if defined explicitly. I remove this code in the new commit to aline with CLI.